samples/dubbogo/multi/server/app/student.go

```diff
@@ -0,0 +1,271 @@
+		return false
+	}
+
+	_, ok := db.nameIndex[name]
+	if ok {
+		return true
+	}
+
+	return false
+}
+
+func (db *StudentDB) existCode(code int64) bool {
+	if code <= 0 {
+		return false
+	}
+
+	_, ok := db.codeIndex[code]
+	if ok {
+		return true
+	}
+
+	return false
+}
+
+// Student student obj.
+type Student struct {
+	ID   string    `json:"id,omitempty"`
+	Code int64     `json:"code,omitempty"`
+	Name string    `json:"name,omitempty"`
+	Age  int32     `json:"age,omitempty"`
+	Time time.Time `json:"time,omitempty"`
+}
+
+// StudentProvider the dubbo provider.
+// like: version: 1.0.0 group: test
+type StudentProvider struct {
+}
+
+// CreateStudent new Student, Proxy config POST.
+func (s *StudentProvider) CreateStudent(ctx context.Context, student *Student) (*Student, error) {
+	println("Req CreateStudent data:%#v", student)
+	if student == nil {
+		return nil, errors.New("not found")
+	}
+	_, ok := studentCache.GetByName(student.Name)
+	if ok {
+		return nil, errors.New("data is exist")
+	}
+
+	b := studentCache.Add(student)
+	if b {
+		return student, nil
+	}
+
+	return nil, errors.New("add error")
+}
+
+// GetStudentByName query by name, single param, Proxy config GET.
+func (s *StudentProvider) GetStudentByName(ctx context.Context, name string) (*Student, error) {
+	println("Req GetStudentByName name:%#v", name)
+	r, ok := studentCache.GetByName(name)
+	if ok {
+		println("Req GetStudentByName result:%#v", r)
+		return r, nil
+	}
+	return nil, nil
+}
+
+// GetStudentByCode query by code, single param, Proxy config GET.
+func (s *StudentProvider) GetStudentByCode(ctx context.Context, code int64) (*Student, error) {
+	println("Req GetStudentByCode name:%#v", code)
+	r, ok := studentCache.GetByCode(code)
+	if ok {
+		println("Req GetStudentByCode result:%#v", r)
+		return r, nil
+	}
+	return nil, nil
+}
+
+// GetStudentTimeout query by name, will timeout for proxy.
+func (s *StudentProvider) GetStudentTimeout(ctx context.Context, name string) (*Student, error) {
+	println("Req GetStudentByName name:%#v", name)
+	// sleep 10s, proxy config less than 10s.
+	time.Sleep(10 * time.Second)
+	r, ok := studentCache.GetByName(name)
+	if ok {
+		println("Req GetStudentByName result:%#v", r)
+		return r, nil
+	}
+	return nil, nil
+}
+
+// GetStudentByNameAndAge query by name and age, two params, Proxy config GET.
+func (s *StudentProvider) GetStudentByNameAndAge(ctx context.Context, name string, age int32) (*Student, error) {
+	println("Req GetStudentByNameAndAge name:%s, age:%d", name, age)
+	r, ok := studentCache.GetByName(name)
+	if ok && r.Age == age {
+		println("Req GetStudentByNameAndAge result:%#v", r)
+		return r, nil
+	}
+	return r, nil
+}
+
+// UpdateStudent update by Student struct, my be another struct, Proxy config POST or PUT.
+func (s *StudentProvider) UpdateStudent(ctx context.Context, Student *Student) (bool, error) {
+	println("Req UpdateStudent data:%#v", Student)
+	r, ok := studentCache.GetByName(Student.Name)
+	if ok {
```

Comment:
hey, we should write elegant codes. the code should be :

if !ok {
}

other codes